### PR TITLE
Thomas/rotate velocity aware box

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
I've added the function suggested by Bo with the option `rotationZAnticlockwiseFromX`. 
Let me know what you think.

### CODE REVIEW

**Comment by Thomas-Ulrich** [comment]
Input file:
```
<freeSurface>9</freeSurface>
<dynamicRupture>8</dynamicRupture>
<absorbing>10,11,12,13,15</absorbing>
<surfaceMSize value="200">8</surfaceMSize>
<area_AspectRatio value="6"/>
<vol_AspectRatio value="12"/>
<SurfaceMeshing SmoothingLevel="1" SmoothingType="Gradient" DiscreteAngle="15.0" Snap="0"></SurfaceMeshing>
<VolumeMeshing  SmoothingLevel="1" SmoothingType="Gradient" SetOptimisation="1"></VolumeMeshing>
<gradation value="0.3"/>
<VelocityAwareMeshing easiFile="RFV_rhomulambda.yaml" elementsPerWaveLength="4">
    <VelocityRefinementCuboid frequency="0.5" centerX="0" centerY="0" centerZ="-15e3"
                              halfSizeX="40e3" halfSizeY="15e3" halfSizeZ="15e3" bypassFindRegionAndUseGroup="1" rotationZAnticlockwiseFromX="30.0"/>
    <VelocityRefinementCuboid frequency="0.25" centerX="0" centerY="0" centerZ="-15e3"
                              halfSizeX="40e10" halfSizeY="15e10" halfSizeZ="15e10" bypassFindRegionAndUseGroup="1"/>
</VelocityAwareMeshing>
````
associated log:
```
Thu Mar 10 20:53:42, Info:  Using SimModSuite
Thu Mar 10 20:53:42, Info:  Loading model
Thu Mar 10 20:53:52, Info:  Extracting cases
Thu Mar 10 20:53:52, Info:  surface smoothing option: surfaceSmoothingLevel surfaceSmoothingType surfaceFaceRotationLimit Snap 1   1   15   0
Thu Mar 10 20:53:52, Info:  volume smoothing option: volumeSmoothingLevel volumeSmoothingType 1   1
Thu Mar 10 20:53:52, Info:  Activating velocity aware meshing, using 4 elements per wavelength and easi file RFV_rhomulambda.yaml
Thu Mar 10 20:53:52, Info:  Adding velocity aware refinement region targeting 0.5 Hz, centered at x = 0 y= 0 z= -15000 with half sizes x = 40000 y = 15000 z = 15000
Thu Mar 10 20:53:52, Info:  rotated around z axis by  30 degree(s) counterclockwise from x axis.
Thu Mar 10 20:53:52, Info:  bypass findRegion and use group = 1
Thu Mar 10 20:53:52, Info:  Adding velocity aware refinement region targeting 0.25 Hz, centered at x = 0 y= 0 z= -15000 with half sizes x = 4e+11 y = 1.5e+11 z = 1.5e+11
Thu Mar 10 20:53:52, Info:  bypass findRegion and use group = 1
Thu Mar 10 20:53:52, Info:  Setting cases
Thu Mar 10 20:53:52, Info:  faceBound[ 9 ] = 1
Thu Mar 10 20:53:52, Info:  faceBound[ 8 ] = 3
Thu Mar 10 20:53:52, Info:  faceBound[ 10 ] = 5
Thu Mar 10 20:53:52, Info:  faceBound[ 11 ] = 5
Thu Mar 10 20:53:52, Info:  faceBound[ 12 ] = 5
Thu Mar 10 20:53:52, Info:  faceBound[ 13 ] = 5
Thu Mar 10 20:53:52, Info:  faceBound[ 15 ] = 5
Thu Mar 10 20:53:52, Info:  face id: 8 , MSize = 200
Thu Mar 10 20:53:52, Info:  Enabling velocity aware meshing
Thu Mar 10 20:53:52, Info:  Gradation rate = 0.3
Thu Mar 10 20:53:52, Info:  Target equivolume AspectRatio = 12
Thu Mar 10 20:53:52, Info:  Target equiarea AspectRatio = 6
Thu Mar 10 20:53:52, Info:  Starting the surface mesher
Thu Mar 10 20:53:52, Info:  Progress: Surface Meshing
Thu Mar 10 20:54:49, Info:  Progress: Adapting Mesh
Thu Mar 10 20:56:21, Info:  Progress: Adapting Mesh
Thu Mar 10 20:56:29, Info:  Progress: Surface Mesh Improver
Thu Mar 10 20:56:29, Info:  Progress: Surface Mesh Improvement , 0 / 2
Thu Mar 10 20:56:29, Info:  Progress: Surface Mesh Improvement , 1 / 2
(...)
```
generated mesh (view from above):
rotateVelocityAware

File: src/input/MeshAttributes.cpp [hunk 1]
```diff
@@ -206,6 +213,10 @@ void MeshAttributes::set_velocity_aware_meshing() {
                           << "with half sizes"
                           << "x =" << cuboid.halfSize[0] << "y =" << cuboid.halfSize[1]
                           << "z =" << cuboid.halfSize[2];
+      if (abs(cuboid.rotationZ) > 0.0) {
```
Comment (krenzland) [inline src/input/MeshAttributes.cpp]: Should be std::abs

**Review by krenzland — CHANGES_REQUESTED** [review]
Looks great, found just one small thing

File: src/input/MeshAttributes.cpp (hunk 1 above)
Comment (krenzland) [reply src/input/MeshAttributes.cpp]: See here for reasoning: https://stackoverflow.com/a/21393170
Using abs can lead to compatibility problems

**Review by krenzland — APPROVED** [review]
Thanks!

**Pull request merged**